compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -369,6 +369,7 @@ pub fn compile_declarative_macro(
     features: &Features,
     def: &ast::Item,
     edition: Edition,
+    is_definition: bool,
```

Comment:
You don't really have to pass this flag from the outside, a "foreign" macro is detectable using the `node_id != DUMMY_NODE_ID` condition, some other lints already use it (try searching for `DUMMY_NODE_ID`).

Reply:
Thanks!